sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/EventHubConsumerClient.cs

```diff
@@ -391,8 +388,10 @@ public virtual async Task<string[]> GetPartitionIdsAsync(CancellationToken cance
 
             try
             {
-                eventChannel = CreateEventChannel((BackgroundPublishReceiveBatchSize * 4));
-                cancelPublishingAsync = await PublishPartitionEventsToChannelAsync(partitionId, startingPosition, options.TrackLastEnqueuedEventProperties, options.OwnerLevel, eventChannel, cancellationSource).ConfigureAwait(false);
+                var channelSize = options.CacheEventCount * 4L;
```

Comment:
This method is due to be restructured to not make use of the background publishing channel before release; the overflow protection here is temporary.